index.html

```diff
@@ -17971,6 +17982,13 @@ <h2 id="AudioWorklet-concepts">
             </figcaption>
           </figure>
           <p>
+            Each <a>BaseAudioContext</a> possesses exactly one
+            <a>AudioWorklet</a>. Scripts imported using this
+            <a>AudioWorklet</a> are run in one or more
```

Comment:
"one or more"

Is this correct?

```js
contextA.worklet.addModule('a.js');
contextB.worklet.addModule('b.js');
```

A processor defined in `a.js` will not be available for `contextB`. I guess AudioWorklet in L17987 means a "general AudioWorklet", but I find a bit confusing. I think we should clarify there is one-to-one mapping between a context and a worklet global scope.

Reply:
Actually I think it’s 1:N for contexts and scopes since we may want to ensure some indeterminacy wrt where rendering work is done to avoid erroneous caching. 

AudioWorklet here should really say “Worklet” since there’s no AudioWorklet interface (applies elsewhere also).

Reply:
> Actually I think it’s 1:N for contexts and scopes since we may want to ensure some indeterminacy wrt where rendering work is done to avoid erroneous caching.

I am confused. Is this something we discussed in person?

> AudioWorklet here should really say “Worklet” since there’s no AudioWorklet interface (applies elsewhere also).

I agree, but that actually opens up another editorial issue. We have a whole section named "the AudioWorklet interface".

Reply:
In a recent meeting the group did discuss the benefits of having more than one scope per context to avoid developers unintentionally saving state in global variables and counting on it being there. Apparently PaintWorklets always use 2 scopes and randomly switch between them. However, we do need to make sure that any given AWP always runs in the same scope, in order to be able to reliably maintain its state -- think filters, delay lines, etc. So... what is this indeterminacy buying us really?

OTOH it's also possible that in the future we might support more than one rendering thread per AudioContext; that was discussed quite a long time ago. This would also result in >1 AWGS per context.

I'd appreciate hearing from other WG folks on these points since we should try to get this right. Certainly at the moment in practice impls will do a 1:1 relationship, it's just a question of futureproofing the spec.

Reply:
I think 1:N mapping between the main and the worklet global scope only makes sense for PaintWorklet case. PaintWorklet is more about supporting "state-less function call" whereas AudioWorkletProcessor is a complete class structure with various states inside. We need to support "stateful" programming model, thus maintaining a consistent global scope is imperative.